src/daft-core/src/array/ops/sketch_quantile.rs

```diff
@@ -0,0 +1,47 @@
+use common_error::DaftError;
+
+use crate::datatypes::DataType;
+use crate::datatypes::Field;
+use crate::utils::sketch::Sketch;
+use crate::{array::DataArray, datatypes::DaftArrowBackedType, datatypes::Float64Array};
+
+use arrow2::array::Array;
+use arrow2::array::BinaryArray;
+use arrow2::array::PrimitiveArray;
+use common_error::DaftResult;
+
+impl<T: DaftArrowBackedType> DataArray<T>
+where
+    T: DaftArrowBackedType,
+{
+    pub fn sketch_quantile(&self, q: &Float64Array) -> DaftResult<Float64Array> {
+        if q.len() != 1 || q.get(0).is_none() {
+            return Err(DaftError::ValueError(String::from(
+                "q parameter of sketch_quantile must be defined",
+            )));
+        }
+
+        let quantile = q.get(0).unwrap();
```

Comment:
We can probably just have this do `q.get(0).expect("q parameter of sketch_quantile must have one non-null element")` and skip the validations above, since we have a strong assumption here for this function and wouldn't expect this error to be caught further up the callstack.